shell/platform/embedder/fixtures/main.dart

```diff
@@ -879,3 +879,22 @@ void nativeArgumentsCallback(List<String> args) native 'NativeArgumentsCallback'
 void dart_entrypoint_args(List<String> args) {
   nativeArgumentsCallback(args);
 }
+
+void snapshotDimensionsCallback(int width, int height) native 'SnapshotDimensionsCallback';
+
+@pragma('vm:entry-point')
+void snapshot_large_scene() {
+  ReceivePort port = ReceivePort();
+  port.listen((dynamic max_size) async {
+    int size = max_size as int;
+
+    PictureRecorder recorder = PictureRecorder();
+    Canvas canvas = Canvas(recorder, Rect.fromLTWH(0, 0, (size * 2).toDouble(), size.toDouble()));
+    final Picture picture = recorder.endRecording();
+    // Set width to double the max size, which will result in height being half the max size after scaling.
+    final image = await picture.toImage(size * 2, size);
```

Comment:
And this would have been a null handle before right?

Reply:
On the Dart side a `NoSuchMethodError` would actually get thrown while constructing the public wrapper `Image` due to the native callback feeding in a null `_Image`.